### ISSUE TITLE
False positive when constructor possibly raises error

### ISSUE THREAD

**Issue by TotalVerb** [tonyhffong/Lint.jl#137]
``` julia
julia> lintstr("""
       immutable Test
           Test(x) = x ? new() : error("constructor must be true")
       end
       """)
1-element Array{Lint.LintMessage,1}:
 none:2 E611 : constructor doesn't seem to return the constructed object
```

this is a common idiom for checking invariants
